src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -0,0 +1,177 @@
+                (m.IsKind(SymbolKind.Property) || m.IsKind(SymbolKind.Field)) &&
+                !m.IsImplicitlyDeclared);
+
+            // Filter out those members that have already been typed
+            var alreadyTestedMembers = GetTestedMembers(semanticModel.SyntaxTree, position, cancellationToken);
+            var untestedMembers = members.Where(m => !alreadyTestedMembers.Contains(m.Name));
+
+            untestedMembers = untestedMembers.Where(m => m.IsEditorBrowsable(document.ShouldHideAdvancedMembers(), semanticModel.Compilation));
+
+            foreach (var untestedProperty in untestedMembers)
+            {
+                context.AddItem(SymbolCompletionItem.CreateWithSymbolId(
+                    displayText: untestedProperty.Name,
+                    insertionText: null,
+                    symbols: ImmutableArray.Create(untestedProperty),
+                    contextPosition: location.SourceSpan.Start,
+                    rules: s_rules));
+            }
+        }
+
+        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
+
+        private static readonly CompletionItemRules s_rules = CompletionItemRules.Create(enterKeyRule: EnterKeyRule.Never);
+
+        internal override bool IsInsertionTrigger(SourceText text, int characterPosition, OptionSet options)
+        {
+            return CompletionUtilities.IsTriggerCharacter(text, characterPosition, options) || text[characterPosition] == ' ';
+        }
+
+        protected (ITypeSymbol type, Location location) GetPatternType(
+            Document document, SemanticModel semanticModel, int position, CancellationToken cancellationToken)
+        {
+            var tree = semanticModel.SyntaxTree;
+            if (tree.IsInNonUserCode(position, cancellationToken))
+            {
+                return default;
+            }
+
+            var token = tree.FindTokenOnLeftOfPosition(position, cancellationToken);
+            token = token.GetPreviousTokenIfTouchingWord(position);
+
+            if (token.Kind() != SyntaxKind.CommaToken && token.Kind() != SyntaxKind.OpenBraceToken)
+            {
+                return default;
+            }
+
+            if (token.Parent?.IsKind(SyntaxKind.PropertySubpattern) == false ||
+                token.Parent.Parent?.IsKind(SyntaxKind.PropertyPattern) == false)
+            {
+                return default;
+            }
+
+            // is Goo { $$
+            // is Goo { P1: 0, $$
+            var propertyPattern = (PropertyPatternSyntax)token.Parent.Parent;
+            var patternType = propertyPattern.Type;
+            if (patternType != null)
+            {
+                var typeSymbol = (ITypeSymbol)semanticModel.GetSymbolInfo(patternType, cancellationToken).Symbol;
```

Comment:
can this be an alias symbol instead?

Reply:
Great, thanks. Is it also true that this will never return anything other than a type in some error cases?